src/components/DocumentationBrowser/DocumentationBrowser.tsx

```diff
@@ -26,10 +26,51 @@ import {
   fetchActionProviders,
   type ActionProviderEntry,
   BOOTSTRAPPED,
+  type ActionProviderState,
 } from "./library";
 import { useEffect, useRef, useState } from "react";
 
-const ActionProviderItem = ({ ap }: { ap: ActionProviderEntry }) => {
+const transformActionProvider = (
+  ap: ActionProviderEntry,
+): ActionProviderState => {
+  const Parameters: Record<string, unknown> = {};
+  const transformed: ActionProviderState = {
+    Type: "Action",
+    ActionUrl: ap.url,
+    Parameters,
+  };
+
+  // Add some, preferably descriptive, info about the available
+  // parameters to the `Parameters` object.
+  if (
+    ap.definition?.input_schema &&
+    typeof ap.definition.input_schema === "object" &&
+    !Array.isArray(ap.definition.input_schema) &&
+    "properties" in ap.definition.input_schema &&
+    ap.definition.input_schema?.properties
+  ) {
+    Object.entries(ap.definition.input_schema.properties).forEach(
+      ([key, value]) => {
+        Parameters[key] =
+          value?.const ?? value?.default ?? value?.description ?? value?.type;
+      },
+    );
+  }
```

Comment:
This is a questionable enhancement, I've added these changes in a separate commit so it can be dropped easily. On one hand, it adds _some_ hinting to the user about what should be in these objects, on the other hand it could easily just be confusing about what's actually being added. Maybe should just be from const or default?

Reply:
Working through this I think this expands this task quite a bit. For example, we've gotta have the AP definitions so do we now load those at app start. Where do we store them, how do we parse them, etc. So I think we should push this into another task if it's desired that we use the definition input schema.